contracts/modules/assessment/Assessment.sol

```diff
@@ -2,515 +2,365 @@
-      require(!hasAlreadyVotedOn[msg.sender][assessmentId], AlreadyVoted());
-      hasAlreadyVotedOn[msg.sender][assessmentId] = true;
-    }
+  /// @notice Returns the assessor group ID for a given claim
+  /// @param claimId The claim identifier
+  /// @return The group ID of the assessors for the claim
+  function assessorGroupOf(uint claimId) override external view returns (uint32) {
+    Assessment memory assessment = _assessments[claimId];
+    require(assessment.start != 0, InvalidClaimId());
 
-    uint96 stakeAmount = stakeOf[msg.sender].amount;
-    require(stakeAmount > 0, StakeRequired());
+    return assessment.assessingGroupId;
+  }
+
+  function getAssessment(uint claimId) override external view returns(Assessment memory assessment) {
+    return _assessments[claimId];
+  }
 
-    Poll memory poll = assessments[assessmentId].poll;
-    require(block.timestamp < poll.end, VotingClosed());
-    require(poll.accepted > 0 || isAcceptVote, AcceptVoteRequired());
+  function getAssessmentResult(uint claimId) override external view returns(uint cooldownEnd, AssessmentStatus status) {
+    Assessment memory assessment = _assessments[claimId];
+    cooldownEnd = assessment.votingEnd + assessment.cooldownPeriod;
+    return (cooldownEnd, _getAssessmentStatus(assessment));
+  }
 
-    if (poll.accepted == 0) {
-      // Reset the poll end date on the first accept vote
-      poll.end = uint32(block.timestamp + MIN_VOTING_PERIOD);
+  function _getAssessmentStatus(Assessment memory assessment) internal view returns(AssessmentStatus status) {
+    if (block.timestamp < assessment.votingEnd) {
+      return AssessmentStatus.VOTING;
     }
 
-    // Check if poll ends in less than 24 hours
-    if (poll.end - block.timestamp < SILENT_ENDING_PERIOD) {
-      // Extend proportionally to the user's stake but up to 1 day maximum
-      poll.end += uint32(
-        Math.min(
-          SILENT_ENDING_PERIOD,
-          SILENT_ENDING_PERIOD * uint(stakeAmount) / (uint(poll.accepted + poll.denied))
-        )
-      );
+    if (!_hasCooldownPassed(assessment)) {
+      return AssessmentStatus.COOLDOWN;
     }
 
-    if (isAcceptVote) {
-      poll.accepted += stakeAmount;
+    if (assessment.acceptVotes > assessment.denyVotes) {
+      return AssessmentStatus.ACCEPTED;
+    } else if (assessment.acceptVotes < assessment.denyVotes) {
+      return AssessmentStatus.DENIED;
     } else {
-      poll.denied += stakeAmount;
+      return AssessmentStatus.DRAW;
     }
+  }
 
-    assessments[assessmentId].poll = poll;
+  /// @notice Returns the ballot for a given claim and assessor
+  /// @param claimId The claim identifier
+  /// @param assessor The address of the assessor
+  /// @return The Ballot struct for the assessor on the claim
+  function ballotOf(uint claimId, address assessor) override external view returns (Ballot memory) {
+    (uint assessorMemberId, ) = _validateAssessor(claimId, assessor);
+    return _ballots[assessorMemberId][claimId];
+  }
```

Comment:
1. I really don't like that we query by assessor address here. If they switch to a different one we can no longer fetch this?
2. It's not a good idea to use `_validateAssessor` here. If the assessor was removed from the group you won't be able to fetch the ballot anymore

I would generally advise to remove the `_validateAssessor` helper function as it does too many things:
- fetches the member id
- verifies it's indeed inside the assessment group
- fetches the assessment itself
- validates the assessment exists
- returns the assessment and the member id
Did I expect it to do this based on its name? Hell no.

Separately I would also inline `_hasCooldownPassed` just because it can be a one-liner so there aren't any "savings"